ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -477,6 +477,14 @@ object Trees {
     final val UnwrapFromThrowable = 30
     final val Throw = 31
 
+    // Floating point bit manipulation, introduced in 1.20
+    final val Float_toBits = 32
+    // final val Float_toRawBits = 33 // Reserved
```

Comment:
Just out of curiosity, I believe the difference between `doubleToRawBits` and `doubleToBits` is how they handle NaN, whether they convert a given NaN to an integer using its bit pattern (raw), or if they use the bit pattern of a "normalized" NaN (like the result of 0.0 / 0.0).

If I understand correctly, are there any use cases for the `toRawBits` version?

---

Also, is it correct that "raw" NaNs typically being caused only from calculations, and as a rule of thumb, we should use normalized NaN (0.0/0.0) when we use NaNs as constants?

Reply:
Yes, that's correct. The specific normalized bit pattern is even spelled out in the JavaDoc.

The typical use case is if you want to implement "NaN-boxing", by storing actual data in the various payloads of NaNs. But IMO a much safer way to do that is to manipulate the `Long`s everywhere, and only bit-concert to `Double` when you know it's a number. 

I have yet to see a compelling use case for the raw variants, TBH.